src/main/java/de/doubleslash/keeptime/view/SettingsController.java

```diff
@@ -531,4 +582,75 @@ private void showLicense(final Licenses license) {
          alert.show();
       }
    }
+
+   private void handleApiOff() {
+      setWebApplicationType("none");
+   }
+
+   private void handleApiOn() {
+      username = authName.getText();
+      password = authPassword.getText();
+
+      createAndSaveUser(username, password);
+
+      Map<String, String> propertiesToUpdate = new HashMap<>();
+      propertiesToUpdate.put("spring.main.web-application-type", "");
+      propertiesToUpdate.put("server.port", authPort.getText());
+
+      propertyWrite(propertiesToUpdate);
+   }
+
+   private void setWebApplicationType(String value) {
+      propertyWrite("spring.main.web-application-type", value);
+   }
+
+   private void createAndSaveUser(String username, String password) {
+      User user = new User();
+      Authorities authorities = new Authorities();
+
+      userRepository.deleteAll();
+      authoritiesRepository.deleteAll();
+
+      user.setUserName(username);
+      user.setPassword("{noop}" + password);
+      user.setEnabled(true);
+
+      authorities.setUserName(username);
+      authorities.setAuthority("ROLE_USER");
+
+      userRepository.save(user);
+      authoritiesRepository.save(authorities);
+   }
+
+   private void propertyWrite(String key, String value) {

```

Comment:
could you test if we can only persist the properties which we are also changing here? so that the .properties file created will only contain at max the two properties
- port
- server mode

I would like that more, than having also all our internal properties exposed in that file.